modules/exceptions.py

```diff
@@ -0,0 +1,2 @@
+class ApiResponseError(Exception):
```

Comment:
You collapsed this file down to declaring one single type? I do think it's a good idea to keep a few different types, I was just suggesting that you have a unified base-type so we could check for any type of error coming from this layer. e.g.:

``` python
class ApiError(Exception):
    """
    Base exception for any errors in the API layer
    """

    pass


class ApiTimeoutError(ApiError):
    """
    Communication with the API timed out
    """

    pass


class ApiResponseDecodeError(ApiError):
    """
    We failed to properly decode the response from the API
    """

    pass


class ApiResponseError(ApiError):
    """
    The API responded with an error
    """

    def __init__(self, message, status_code):
        self.status_code = status_code
        return super().__init__(message)


class ApiResponseErrorList(ApiResponseError):
    """
    The API responded with a list of errors,
    which are included in self.errors
    """

    def __init__(self, message, status_code, errors):
        self.errors = errors
        return super().__init__(message, status_code)
```

Then users of `api.py` can be general or specific when handling errors:

``` python
try:
    api.do_something()
except ApiTimeoutError:
    print('API was super-slow')
except APIError:
    print('API had some other error')
```